### PR TITLE
Implement RFC-487

### PR DESCRIPTION
**Pull request by juancarlospaco**
- See [RFC-487](https://github.com/nim-lang/RFCs/issues/487#issue-1379849715)
- Unrelated CI fail, Azure complains `fatal: unable to access 'https://github.com/nim-lang/nimble.git/': Could not resolve host: github.com` and/or timeout.

### CODE REVIEW

File: lib/pure/httpcore.nim [hunk 1]
```diff
@@ -51,6 +51,58 @@ type
     HttpPatch = "PATCH"      ## Applies partial modifications to a resource.
 
 
+since (1, 7):
+  type PreparedRequest* = tuple[httpMethod: HttpMethod, url: Uri, headers: seq[(string, string)], body: string]
```
Comment (Araq) [inline lib/pure/httpcore.nim]: This screams "Object here instead of tuple!" to me. ;-)

**Comment by Araq** [comment]
There is a much better way to do this...

**Comment by juancarlospaco** [comment]
@Araq Explain the way, here or on the RFC, or suggest something, consider this also works for JS and its lightweight. 😲

File: changelog.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 
 ## Changes affecting backward compatibility
-- `httpclient.contentLength` default to `-1` if the Content-Length header is not set in the response, it followed Apache HttpClient(Java), http(go) and .Net HttpWebResponse(C#) behavior. Previously raise `ValueError`.  
+- `httpclient.contentLength` default to `-1` if the Content-Length header is not set in the response, it followed Apache HttpClient(Java), http(go) and .Net HttpWebResponse(C#) behavior. Previously raise `ValueError`.
```
Comment (Varriount) [inline changelog.md]:
```suggestion
- `httpclient.contentLength` defaults to `-1` if the Content-Length header is not set in the response, it followed Apache HttpClient(Java), http(go) and .Net HttpWebResponse(C#) behavior. Previously raise `ValueError`.
```

**Comment by Varriount** [comment]
@Araq This looks sensible to me, however I'll label this PR with `Requires Araq` - feel free to remove the label if it's ok to merge,

**Comment by Araq** [comment]
Give it a dedicated module and implement it as a single string internally with accessors that extract the data. Maybe it's overkill though. But the current design is effectively un-changable, you simply cannot ever optimize it. For a stdlib module that's a bad idea.

**Pull request closed without merging**